File: ios-native/CommandComms/Audio/SignalQuality.swift
import Foundation

enum SignalQuality: String {
    case none = "NONE"
    case excellent = "EXCELLENT"
    case good = "GOOD"
    case fair = "FAIR"
    case poor = "POOR"

    static func classify(lossPct: Double, jitterMs: Double, framesInWindow: Int) -> SignalQuality {
        if framesInWindow < 10 { return .none }
        if lossPct < 2.0 && jitterMs < 15.0 { return .excellent }
        if lossPct < 8.0 && jitterMs < 30.0 { return .good }
        if lossPct < 20.0 && jitterMs < 60.0 { return .fair }
        return .poor
    }
}

File: android-native/app/src/main/java/com/reedersystems/commandcomms/audio/radio/SignalQuality.kt
package com.reedersystems.commandcomms.audio.radio

enum class SignalQuality {
    NONE,
    EXCELLENT,
    GOOD,
    FAIR,
    POOR;

    companion object {
        fun classify(lossPct: Double, jitterMs: Double, framesInWindow: Long): SignalQuality {
            if (framesInWindow < 10) return NONE
            return when {
                lossPct < 2.0 && jitterMs < 15.0 -> EXCELLENT
                lossPct < 8.0 && jitterMs < 30.0 -> GOOD
                lossPct < 20.0 && jitterMs < 60.0 -> FAIR
                else -> POOR
            }
        }
    }
}
